hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockOutputStream.java

```diff
@@ -701,10 +704,13 @@ public void cleanup(boolean invalidateClient) {
     xceiverClient = null;
     cleanup();
 
-    if (bufferList !=  null) {
+    if (bufferList != null) {
       bufferList.clear();
     }
     bufferList = null;
+    if (lastChunkBuffer != null) {
+      DIRECT_BUFFER_POOL.returnBuffer(lastChunkBuffer);
```

Comment:
@ashishkumar50 , sorry forget one comment here, let's set lastChunkBuffer to null after it's released.

Reply:
Done